### PR TITLE
Add API Chat feature UI implementation

### PR DESCRIPTION
**Pull request by ashera96**
## Purpose

This PR adds the UI changes related to the API Chat feature.

### CODE REVIEW

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx [hunk 1]
```diff
@@ -0,0 +1,781 @@
...
+    const [inputQuery, setInputQuery] = useState('');
+    const [lastQuery, setLastQuery] = useState('');
+    const [finalOutcome, setFinalOutcome] = useState('');
+    const [executionResults, setExecutionResults] = useState([]);
+    const [isExecutionError, setIsExecutionError] = useState(false);
+    const [isAgentTerminating, setIsAgentTerminating] = useState(false);
+    const [apiAccessToken, setApiTestAccessToken] = useState('');
+
+    useEffect(() => {
+        setApiTestAccessToken('abc');
```
Comment (Krishanx92) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx:109]: abc?

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+        if (isAgentTerminating) {
+            setTimeout(() => {
+                setIsAgentTerminating(false);
+                setIsAgentRunning(false);
+            }, 2000);
+        }
+    }, [isAgentTerminating]);
+
+    useEffect(() => {
+        if (api.id && isApiChatEnabled && isAIFeatureAuthTokenProvided) {
```
Comment (Krishanx92) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx:312]: better to add api && api.id

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/AIGeneratedApiRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    url: 'https://localhost:8243/pizzashack/1.0.0/menu',
+                    credentials: 'same-origin',
+                    headers: {
+                        accept: 'application/json',
+                    },
+                    method: 'GET',
+                },
+                accessToken,
+            );
+            console.log(spec.servers[0].url);
```
Comment (Krishanx92) [inline portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/AIGeneratedApiRequest.tsx:81]: do we need this logs?

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx (hunk 1 above)
Comment (ashera96) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/ApiChat.jsx]: Fixed

File: portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/AIGeneratedApiRequest.tsx (hunk 3 above)
Comment (ashera96) [reply portals/devportal/src/main/webapp/source/src/app/components/Apis/Details/ApiChat/components/AIGeneratedApiRequest.tsx]: This component was deleted

**Pull request merged**